sdk/eventhub/event-hubs/test/eventHubConsumerClient.spec.ts

```diff
@@ -596,5 +596,97 @@ describe("EventHubConsumerClient", () => {
       }
       logTester.assert();
     });
+
+    it("Stops receiving events if close is immediately called, single partition. #RunnableInBrowser", async function(): Promise<
+      void
+    > {
+      const partitionId = "0";
+      const client = new EventHubConsumerClient(
+        EventHubConsumerClient.defaultConsumerGroupName,
+        service.connectionString,
+        service.path
+      );
+
+      clients.push(client);
+
+      let initializeCalled = false;
+      let closeCalled = false;
+
+      const subscription = client.subscribe(partitionId, {
+        async processError() {},
+        async processEvents() {},
+        async processClose() {
+          closeCalled = true;
+        },
+        async processInitialize() {
+          initializeCalled = true;
+        }
+      });
+
+      await subscription.close();
+
+      await loopUntil({
+        maxTimes: 10,
+        name: "Wait for the subscription to stop running.",
+        timeBetweenRunsMs: 100,
+        async until() {
+          return !subscription.isRunning;
+        }
+      });
+
+      if (initializeCalled) {
```

Comment:
To avoid any randomness you might as well just add "&& initializeCalled" to the loopUntil condition.

Reply:
I'll update this to ensure that `initializeCalled` and `closeCalled` are always equal, since that's really want I want to ensure.

In this test, `initializeCalled` should always be false now, but previously it was always true. I don't actually care if `initializeCalled` is true or false as long as `closeCalled` matches.